codegen/src/main/java/software/amazon/awssdk/codegen/IntermediateModelBuilder.java

```diff
@@ -107,12 +107,20 @@ public IntermediateModel build() {
             new HashMap<>(new AddCustomAuthorizers(this.service, getNamingStrategy()).constructAuthorizers());
 
         OperationModel endpointOperation = null;
+        boolean endpointCacheRequired = false;
 
         for (OperationModel o : operations.values()) {
             if (o.isEndpointOperation()) {
                 endpointOperation = o;
-                break;
             }
+
+            if (o.getEndpointDiscovery() != null && o.getEndpointDiscovery().isRequired()) {
+                endpointCacheRequired = true;
```

Comment:
So the "new" behavior is: if any operation requires endpoint discovery, enable it by default?

Reply:
Yes, exactly. I also stated that in the PR description.